### ISSUE TITLE
Build <gaia-fast-list>

### ISSUE THREAD

**Issue by wilsonpage** [gaia-components/fast-list#31]
I suggest we flesh out the initial component in this repo before it flies the nest :)

The component will be more opinionated than the lower level library, containing styling and most of the data-source preconfigured internally.

**Comment by wilsonpage** [comment]
Hopefully we can provide a fully declarative way for the user to provide their template/populate configuration.

``` html
<gaia-fast-list>
  <template section>
    <h2>${section}</h2>
  <template>
  <template item>
    <li><h3>${item.title}</h3><p>${item.body}</p></li>
  <template>
</gaia-fast-list>
```

**Comment by wilsonpage** [comment]
```
<gaia-fast-list>
  <template section>
    <h2 data-bind="textContent:section"></h2>
  <template>
  <template item>
    <li>
      <h3 data-bind="textContent:item.title"></h3>
      <p data-bind="textContent:item.body"></p>
      <img data-bind="src:item.image"/>
    </li>
  <template>
</gaia-fast-list>
```

**Comment by wilsonpage** [comment]
Maybe trickier to parse, but mustache style is more conventional.

``` html
<gaia-fast-list>
  <template section>
    <h2>{{textContent.section}}</h2>
  <template>
  <template item>
    <li>
      <h3>{{textContent.item.title}}</h3>
      <p>{{item.body}}</p>
      <img src="{{item.image}}"/>
    </li>
  <template>
</gaia-fast-list>
```

**Comment by wilsonpage** [comment]
The above after parse could get rewritten into the `data-bind` style to make is easier to pick out nodes during the populate phase.

**Comment by wilsonpage** [comment]
Going to fix #28 as part of this issue too, as Music depends on list-items being the light-dom so that it can catch `<a>` clicks.
